tools/eslint-rules/only-ascii-characters.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @fileoverview Prohibit the use of non-ascii characters
+ * @author Kalon Hinds
+ */
+
+/* eslint no-control-regex:0 */
+
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const nonAsciiPattern = new RegExp('([^\x00-\x7F])', 'g');
+const suggestions = {
+  '’': '\'',
+  '—': '-'
+};
+
+module.exports = {
+  create: function(context) {
+    return {
+      Program: (node) => {
+        const errors = [];
+        const source = context.getSourceCode();
+        const sourceTokens = source.getTokens(node);
+        const commentTokens = source.getAllComments();
+        const tokens = sourceTokens.concat(commentTokens);
+
+        tokens.forEach((token) => {
+          const { value } = token;
+          const matches = value.match(nonAsciiPattern);
+
+          if (matches) {
+            const { loc } = token;
+
+            // Will only report the first non-ascii character per line
+            const character = matches[0];
```

Comment:
Either the comment is misleading, or it works not the way you planned it to. This will report the first non-ASCII character per token, not per line.